test/functional/rpc_psbt.py

```diff
@@ -655,10 +659,11 @@ def test_psbt_input_keys(psbt_input, keys):
                 break
         psbt_in = dec["inputs"][input_idx]
         # Calculate the input weight
-        # (prevout + sequence + length of scriptSig + 2 bytes buffer) * 4 + len of scriptwitness
+        # (prevout + sequence + length of scriptSig + scriptsig + 1 byte buffer) * WITNESS_SCALE_FACTOR + num scriptWitness stack items + (length of stack item + stack item) * N stack items + 1 byte buffer
         len_scriptsig = len(psbt_in["final_scriptSig"]["hex"]) // 2 if "final_scriptSig" in psbt_in else 0
-        len_scriptwitness = len(psbt_in["final_scriptwitness"]["hex"]) // 2 if "final_scriptwitness" in psbt_in else 0
-        input_weight = ((41 + len_scriptsig + 2) * 4) + len_scriptwitness
+        len_scriptsig += len(ser_compact_size(len_scriptsig)) + 1
+        len_scriptwitness = (sum([(len(x) // 2) + ser_compact_size(len(x) // 2) for x in psbt_in["final_scriptwitness"]]) + len(psbt_in["final_scriptwiness"]) + 1) if "final_scriptwitness" in psbt_in else 0
+        input_weight = ((40 + len_scriptsig) * WITNESS_SCALE_FACTOR) + len_scriptwitness
```

Comment:
```suggestion
        input_weight = (40 + len_scriptsig) * WITNESS_SCALE_FACTOR + len_scriptwitness
```

Reply:
In both tests, any reason to not keep the current value of `41` instead of adding `+ 1` two lines earlier?

grepping-wise, 41 seems to be the usual value
```
src/test/transaction_tests.cpp:915:    t.vin.resize(2438); // size per input (empty scriptSig): 41 bytes
src/wallet/rpc/spend.cpp:574:                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, weight cannot be less than 165 (41 bytes (size of outpoint + sequence + empty scriptSig) * 4 (witness scaling factor)) + 1 (empty witness)");
test/functional/test_framework/script_util.py:27:# Blank Input: 32 [PrevTxHash] + 4 [Index] + 1 [scriptSigLen] + 4 [SeqNo] = 41 bytes
```

Reply:
Thanks